### PR TITLE
[Java] Introduce clustered service specific SnapshotDurationTracker

### PR DESCRIPTION
**Pull request by eliquinox**
Following #1532, this PR introduces snapshot duration tracking to clustered services.
Note that `aeron.cluster.service.snapshot.threshold` property usage sets the threshold value for **all** clustered services. If a per-service threshold setting is needed, `ClusteredServiceContainer.Context#snapshotDurationThresholdNs` API can be used.

### CODE REVIEW

File: aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java [hunk 1]
```diff
@@ -1975,8 +1984,14 @@ void shouldTrackSnapshotDuration()
         cluster.awaitSnapshotCount(1);
         cluster.awaitNeutralControlToggle(leader);
 
-        assertEquals(1L, snapshotDurationThresholdExceededCount.get());
-        assertTrue(maxSnapshotDuration.get() > 0);
+        assertEquals(1L, totalSnapshotDurationTracker.snapshotDurationThresholdExceededCount().get());
+        assertTrue(totalSnapshotDurationTracker.maxSnapshotDuration().get() > 0);
+
+        assertEquals(1L, service1SnapshotDurationTracker.snapshotDurationThresholdExceededCount().get());
+        assertTrue(service1SnapshotDurationTracker.snapshotDurationThresholdExceededCount().get() > 0);
```
Comment (vyazelenko) [inline aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java:1991]: The second `assertTrue` is redundant as it checks the same thing as the previous line.
Did you intend to check the `maxSnapshotDuration` instead?

File: aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java [hunk 2]
```diff
@@ -1975,8 +1984,14 @@ void shouldTrackSnapshotDuration()
...
+
+        assertEquals(1L, service2SnapshotDurationTracker.snapshotDurationThresholdExceededCount().get());
+        assertTrue(service2SnapshotDurationTracker.snapshotDurationThresholdExceededCount().get() > 0);
```
Comment (vyazelenko) [inline aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java:1994]: The second `assertTrue` is redundant as it checks the same thing as the previous line.
Did you intend to check the `maxSnapshotDuration` instead?

File: aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java (hunk 1 above)
Comment (eliquinox) [reply aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java]: fixed

File: aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java (hunk 2 above)
Comment (eliquinox) [reply aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java]: fixed

**Comment by vyazelenko** [comment]
@eliquinox The build is broken due to Checkstyle issues. https://github.com/real-logic/aeron/actions/runs/8249191083

File: aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java [hunk 3]
```diff
@@ -1975,8 +1984,14 @@ void shouldTrackSnapshotDuration()
         cluster.awaitSnapshotCount(1);
         cluster.awaitNeutralControlToggle(leader);
 
-        assertEquals(1L, snapshotDurationThresholdExceededCount.get());
-        assertTrue(maxSnapshotDuration.get() > 0);
+        assertEquals(1L, totalSnapshotDurationTracker.snapshotDurationThresholdExceededCount().get());
+        assertTrue(totalSnapshotDurationTracker.maxSnapshotDuration().get() > 0);
+
+        assertEquals(1L, service1SnapshotDurationTracker.snapshotDurationThresholdExceededCount().get());
+        assertTrue(service1SnapshotDurationTracker.maxSnapshotDuration().get() > 0);
```
[the remark this answers is not in the document]
Comment (eliquinox) [reply aeron-system-tests/src/test/java/io/aeron/cluster/ClusterTest.java]: Good idea. Asserting on the lower limits of specific durations now.

**Pull request merged**